util/chunk/row.go

```diff
@@ -110,27 +97,23 @@ func (r Row) getNameValue(colIdx int) (string, uint64) {
 
 // GetEnum returns the Enum value with the colIdx.
 func (r Row) GetEnum(colIdx int) types.Enum {
-	name, val := r.getNameValue(colIdx)
-	return types.Enum{Name: name, Value: val}
+	return r.c.columns[colIdx].GetEnum(r.idx)
 }
 
 // GetSet returns the Set value with the colIdx.
 func (r Row) GetSet(colIdx int) types.Set {
-	name, val := r.getNameValue(colIdx)
-	return types.Set{Name: name, Value: val}
+	return r.c.columns[colIdx].GetSet(r.idx)
 }
 
 // GetMyDecimal returns the MyDecimal value with the colIdx.
 func (r Row) GetMyDecimal(colIdx int) *types.MyDecimal {
-	col := r.c.columns[colIdx]
-	return (*types.MyDecimal)(unsafe.Pointer(&col.data[r.idx*types.MyDecimalStructSize]))
+	v := r.c.columns[colIdx].GetDecimal(r.idx)
```

Comment:
Can we change `GetDecimal` to return pointer?

Reply:
fixed.